pulsar-transformations/src/main/resources/config-schema.yaml

```diff
@@ -141,6 +142,57 @@ components:
               description: The delimiter to use when concatenating the field names.
               default: _
         - "$ref": "#/components/schemas/Part"
+    ComputeFields:
+      allOf:
+        - "$ref": "#/components/schemas/Step"
+        - type: object
+          description: Adds or updates fields dynamically based on an expression (Currently only AVRO is supported).
+          properties:
+            type:
+              type: string
+              enum:
+                - compute-fields
+            fields:
+              type: [ array, 'null' ]
+              items:
+                type:
+                  - object
+                minLength: 1
+                properties:
+                  name:
+                    type: string
+                    description: The name of the field to compute. If the field name already exist, it will be overwritten.
+                  expression:
+                    type: string
+                    description: An EL-like expression to compute the field value. It can refer to other attributes in the record.
+                  type:
+                    type: string
+                    description: The data type of the computed fields. It should match with the output of the expression.
+                    enum:
+                      - STRING
+                      - INT32
+                      - INT64
+                      - FLOAT
+                      - DOUBLE
+                      - BOOLEAN
+                  part:
+                    type: string
+                    description: The part of the message to apply the computed fields to. If left empty, the fields will apply to the `value`.
```

Comment:
For the other transforms, we apply on both key and value if part is empty. Shouldn't we do the same for consistency ?

Reply:
Yeah check the comment above, from UX, when computing a field, generally I don't think the same field need to be created in both parts. In this case, I'm preferring usability over consistency.